### PR TITLE
Ensure execution_timeout as timedelta

### PR DESCRIPTION
**Pull request by pingzh**
so that it can be correctly serialized and deserialized

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/models/baseoperator.py [hunk 1]
```diff
@@ -767,7 +767,13 @@ def __init__(
         self.email = email
         self.email_on_retry = email_on_retry
         self.email_on_failure = email_on_failure
+
+        if execution_timeout is not None and not isinstance(execution_timeout, timedelta):
```
Comment (kaxil) [inline airflow/models/baseoperator.py:771]:
```suggestion
        if not isinstance(execution_timeout, timedelta):
```
should be enough as we already have the default set in ctor:

https://github.com/apache/airflow/blob/d4d29718fdc46cd49e8b5be2e54fb313d595f06e/airflow/models/baseoperator.py#L205

Comment (kaxil) [reply airflow/models/baseoperator.py:771]: Even if you have plan to allow None, the following should work:

```
if execution_timeout and not isinstance(execution_timeout, timedelta):
```

Comment (pingzh) [reply airflow/models/baseoperator.py:771]: nice, updated.

**Review by kaxil — APPROVED** [review]

**Comment by pingzh** [comment]
@kaxil do you think we can merge this PR? i have been force pushed several times.but CI somehow is stuck and got `cancelled checks` for all of them. and i believe it should not be related to this PR. 

thanks.

File: airflow/models/baseoperator.py [hunk 2]
```diff
@@ -767,7 +767,13 @@ def __init__(
         self.email = email
         self.email_on_retry = email_on_retry
         self.email_on_failure = email_on_failure
+
+        if execution_timeout and not isinstance(execution_timeout, timedelta):
```
Comment (uranusjr) [inline airflow/models/baseoperator.py]:
```suggestion
        if execution_timeout is not None and not isinstance(execution_timeout, timedelta):
```

**Review by uranusjr — APPROVED** [review]
Let’s try CI again.

**Review by KevinYang21 — APPROVED** [review]

**Comment by kaxil** [comment]
🎉

**Pull request merged**